### PR TITLE
Implement pagination, sort and swagger

### PR DESCRIPTION
**Pull request by chertiav**
- added PageResponse and PageMetadata DTOs.

### CODE REVIEW

File: src/main/java/com/achdev/onlinebookstoreapp/controller/BookController.java [hunk 1]
```diff
@@ -25,8 +27,9 @@ public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public PageResponse<BookDto> getAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/controller/BookController.java:50]: Add Swagger annotations for each endpoint

Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/controller/BookController.java:50]: You can return just Page, not PageResponse

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/com/achdev/onlinebookstoreapp/controller/BookController.java (hunk 1 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/controller/BookController.java:50]: Ok

Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/controller/BookController.java:50]: The PageResponse is custom, and looks like : 
{
  "content": [
    {
      "id": 0,
      "title": "string",
      "author": "string",
      "isbn": "string",
      "price": 0,
      "description": "string",
      "coverImage": "string"
    }
  ],
  "metadata": {
    "currentPage": 0,
    "pageSize": 0,
    "totalPageCount": 0,
    "totalElementCount": 0
  }
}

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**